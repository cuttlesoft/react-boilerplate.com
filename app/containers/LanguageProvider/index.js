/*
 *
 * LanguageProvider
 *
 * this component connects the redux state language locale to the
 * IntlProvider component and i18n messages (loaded from `app/translations`)
 */

import React, { useContext } from 'react'
import PropTypes from 'prop-types'
import { IntlProvider } from 'react-intl'
import { observer } from 'mobx-react'
import { LocaleStoreContext } from '../../stores/LocaleStore'

export const LanguageProvider = observer(props => {
  const localStore = useContext(LocaleStoreContext)

  return (
    <IntlProvider
      locale={localStore.locale}
      key={localStore.locale}
      messages={props.messages[localStore.locale]}
    >
      {React.Children.only(props.children)}
    </IntlProvider>
  )
})

LanguageProvider.propTypes = {
  locale: PropTypes.string,
  messages: PropTypes.object,
  children: PropTypes.element.isRequired,
}

export default LanguageProvider
