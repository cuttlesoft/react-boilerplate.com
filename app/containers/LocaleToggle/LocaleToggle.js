import React, { useContext } from 'react'

// Components
import { Toggle } from 'components/Toggle'

// Utils, Services & Messages
import messages from './LocaleToggle.messages'
import { appLocales } from '../../i18n'

// Styles
import { Wrapper } from './styles'

// Store
import { LocaleStoreContext } from '../../stores/LocaleStore'

/*
 *
 * LanguageToggle
 *
 */
export const LocaleToggle = () => {
  const localeStore = useContext(LocaleStoreContext)

  return (
    <Wrapper>
      <Toggle
        value={localeStore.locale}
        values={appLocales}
        messages={messages}
        onToggle={evt => localeStore.changeLocale(evt.target.value)}
      />
    </Wrapper>
  )
}

export default LocaleToggle
