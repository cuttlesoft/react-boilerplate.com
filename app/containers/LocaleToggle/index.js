/*
 *
 * LanguageToggle
 *
 */

import React, { useContext } from 'react'

import Toggle from 'components/Toggle'
import messages from './messages'
import { appLocales } from '../../i18n'
import { LocaleStoreContext } from '../../stores/LocaleStore'

import { Wrapper } from './Wrapper'

export const LocaleToggle = () => {
  const localeStore = useContext(LocaleStoreContext)

  return (
    <Wrapper>
      <Toggle
        value={localeStore.locale}
        values={appLocales}
        messages={messages}
        onToggle={evt => localeStore.changeLocale(evt.target.value)}
      />
    </Wrapper>
  )
}

export default LocaleToggle
