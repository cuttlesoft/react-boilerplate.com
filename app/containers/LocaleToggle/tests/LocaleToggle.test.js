import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'mobx-react'
import { RouterStore } from 'mobx-react-router'

// Components
import LocaleToggle from '../LocaleToggle'
import { LanguageProvider } from '../../LanguageProvider'

// Utils, Services & Messages
import { translationMessages } from '../../../i18n'

// Stores
import trunk from '../../../configureStore'

describe('LocaleToggle', () => {
  let stores

  beforeAll(() => {
    const routingStore = new RouterStore()

    stores = {
      routing: routingStore,
    }

    trunk.init()
  })

  it('should match the snapshot', () => {
    const { container } = render(
      <Provider {...stores}>
        <LanguageProvider messages={translationMessages}>
          <LocaleToggle />
        </LanguageProvider>
      </Provider>,
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should present the default `en` english language option', () => {
    const { container } = render(
      <Provider {...stores}>
        <LanguageProvider messages={translationMessages}>
          <LocaleToggle />
        </LanguageProvider>
      </Provider>,
    )

    expect(container.querySelector('option[value="en"]')).not.toBeNull()
  })
})
