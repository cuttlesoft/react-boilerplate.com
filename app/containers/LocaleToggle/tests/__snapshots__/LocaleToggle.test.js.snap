// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleToggle should match the snapshot 1`] = `
<div
  class="LocaleTogglestyles__Wrapper-sc-1ugarqm-0 jxmXgR"
>
  <select
    class="Select-sc-1sm01tk-0 fSKxbD"
  >
    <option
      value="en"
    >
      en
    </option>
    <option
      value="de"
    >
      de
    </option>
  </select>
</div>
`;
